### PR TITLE
Use nats-server crate in tests

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

File: async-nats/tests/client_tests.rs [hunk 1]
```diff
@@ -290,7 +287,7 @@ mod client {
 
     #[tokio::test]
     async fn token_auth() {
-        let server = nats_server::run_server("tests/configs/token.conf");
+        let server = nats_server::run_server("../async-nats/tests/configs/token.conf");
```
Comment (caspervonb) [inline async-nats/tests/client_tests.rs:290]: Test didn't move, so why the change?

Comment (caspervonb) [reply async-nats/tests/client_tests.rs:290]: Ah right because the crate resolves paths, we should resolve them here instead.

Comment (Jarema) [reply async-nats/tests/client_tests.rs]: yeah, thats why. Removed nats-server resolving the path.

**Review by caspervonb — APPROVED** [review]
lgtm
